modules/Kconfig.cmsis

```diff
@@ -22,7 +22,7 @@ endif
 
 menuconfig CMSIS_DSP
 	bool "CMSIS-DSP Library Support"
-	depends on (CPU_CORTEX && NEWLIB_LIBC) || ARCH_POSIX
+	depends on NEWLIB_LIBC || ARCH_POSIX
```

Comment:
We can only do this when all CMSIS-DSP functions (i.e. not just "Basic Math") are verified to be working on non-Cortex.

Reply:
Is there no incremental approach to this? I really don't think we need to go all in right away do we?

Reply:
well, one way to do that would be to add `depends on CPU_CORTEX` to all the other Kconfigs (i.e. other than "Basic Math") in the meantime.

Reply:
Done